sdk/keyvault/Azure.Security.KeyVault.Secrets/src/RecoverDeletedSecretOperation.cs

```diff
@@ -44,33 +42,20 @@ internal RecoverDeletedSecretOperation(KeyVaultPipeline pipeline, Response<Secre
         public override SecretProperties Value => _value;
 
         /// <inheritdoc/>
-        public override bool HasCompleted => _completed;
+        public override bool HasCompleted => _operationInternal.HasCompleted;
 
         /// <inheritdoc/>
         public override bool HasValue => true;
 
         /// <inheritdoc/>
-        public override Response GetRawResponse() => _response;
+        public override Response GetRawResponse() => _operationInternal.RawResponse;
 
         /// <inheritdoc/>
         public override Response UpdateStatus(CancellationToken cancellationToken = default)
         {
-            if (!_completed)
+            if (!HasCompleted)
             {
-                using DiagnosticScope scope = _pipeline.CreateScope($"{nameof(RecoverDeletedSecretOperation)}.{nameof(UpdateStatus)}");
-                scope.AddAttribute("secret", _value.Name);
```

Comment:
You were adding the secret DiagnosticScope attribute before. You can do the same with `OperationInternal` by passing the parameter to `OperationInternal.ctor`

Reply:
Oops. Forgot to add that back. Thanks for the catch!